src/utils/typename.cpp

```diff
@@ -40,12 +40,12 @@ std::string safe_demangle(const char* mangled)
     auto error_str = demangle_errors.find(status);
     if (error_str != demangle_errors.end())
     {
-      throw H2Exception(std::string("Demangling failed: ")
-                        + error_str->second);
+      throw H2NonfatalException(std::string("Demangling failed: ")
```

Comment:
I don't like this. It's not the thrower's job to say something's nonfatal... That's really a decision for the downstream catcher to decide based on information contained in the exception. OTOH, this is really an implementation detail private to this file so I'm disinclined to care all that much.

Reply:
I think `NonfatalException` is the right choice here semantically (failing to demangle something is recoverable if you catch it). I added a catch in the `get_type_name` (which in any case is `namespace internal`) so this effectively never gets seen, mainly because we'd have to scatter `try ... catch` in a bunch of places in the code.

Reply:
Sure but at a "C++ in general" level, arguably one shouldn't bother throwing anything if an error mode is encountered that is actually not recoverable -- that should just terminate (or abort or exit or whatever). Basically: don't bother throwing a `FatalException` and don't bother labeling a `NonfatalException` as nonfatal...